catalog/app/testing/feature.js

```diff
@@ -99,15 +138,23 @@ const renderSteps = (steps, getCtx, next) => {
     return;
   }
 
+  if (steps[0].type === Branch) {
+    const [branch, ...rest] = steps;
+    renderSteps(branch.steps, getCtx);
+    renderSteps(rest, getCtx, next);
+    return;
+  }
+
   if (steps[0].type === Then) {
-    const isNotWhen = ({ type }) => type !== When;
-    const thens = takeWhile(steps, isNotWhen);
-    const rest = dropWhile(steps, isNotWhen);
+    const isThen = ({ type }) => type !== When && type !== Branch;
+    const thens = takeWhile(steps, isThen);
```

Comment:
[partition](https://lodash.com/docs/4.17.10#partition) is a function that expresses your intent here better I think

Reply:
@meffij `partition` does something slightly different: it puts all the elements satisfying the predicate into the first group, and all the others into the second, but i need to split the array at the first occasion of the element satisfying the predicate

Reply:
I see, so you aren't guaranteed to take the whole array.